File: elixir-phoenix/lib/app_web/controllers/hello_controller.ex
defmodule AppWeb.HelloController do
  use AppWeb, :controller

  def index(conn, _params) do
    json(conn, %{message: "Hello from Phoenix!"})
  end

  def greet(conn, %{"name" => name}) do
    json(conn, %{message: "Hello, #{name}!"})
  end

  def greet(conn, _params) do
    json(conn, %{message: "Hello, World!"})
  end
end

File: laravel/app/Http/Controllers/HelloController.php
<?php

namespace App\Http\Controllers;

use Illuminate\Http\JsonResponse;
use Illuminate\Http\Request;

class HelloController extends Controller
{
    public function index(): JsonResponse
    {
        return response()->json(['message' => 'Hello from Laravel!']);
    }

    public function greet(Request $request): JsonResponse
    {
        $validated = $request->validate([
            'name' => 'required|string|max:100',
        ]);

        return response()->json([
            'message' => "Hello, {$validated['name']}!",
        ]);
    }
}
